### PR TITLE
Add support for redirecting including a query

### PR DESCRIPTION
**Pull request by aredridel**

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -1482,6 +1482,36 @@ describe('scheme', function () {
...
+                });
+
+                server.inject('/?test=1', function (res) {
+
+                    expect(res.statusCode).to.equal(302);
+                    expect(res.headers.location).to.equal('http://example.com/login?mode=1&next=%2F%3Ftest%3D1');
+                    done();
+                });
+            });
+        });
```
Comment (hueniverse) [inline test/index.js]: Missing empty line after.

**Comment by hueniverse** [comment]
```
> encodeURIComponent(Url.format('http://example.com/login?mode=1'));
'http%3A%2F%2Fexample.com%2Flogin%3Fmode%3D1'
> encodeURIComponent('http://example.com/login?mode=1');
'http%3A%2F%2Fexample.com%2Flogin%3Fmode%3D1'
```

What am I missing? I'm confused...

**Comment by aredridel** [comment]
Well gee whiz. I mistook where the error is and didn't make sure to make sure my test failed first. Geez.

I was confusing `url.path` with `url.pathname`

**Comment by hueniverse** [comment]
No worries.

**Comment by waywaaard** [comment]
I've some problems getting the redirect to work with multiple query parms that are url encoded.

This is the next URL outputted to the dom:`?next=/run?language=python3&id=testidwhynot&embedType=sourcebox&code=public%20static%20void%20main(String%20args%5B%5D)%20%7B%0A%09System.out.println(%22test%22)%3B%0A%7D`

and when clicking on it, I get redirected to `http://localhost:3000/run?language=python3`. So, the rest of my query params get swallowed.

Is there a solution or a fix for that behavior?

**Pull request closed without merging**